contracts/LifCrowdsale.sol

```diff
@@ -131,17 +150,23 @@ contract LifCrowdsale is Ownable, Pausable {
     // calculate token amount to be created
     uint256 tokens = weiAmount.mul(rate);
 
-    // update state
-    weiRaised = weiRaised.add(weiAmount);
-    purchases[beneficiary] = weiAmount;
-    tokensSold = tokensSold.add(tokens);
+    // check if we are in the public presale block range
+    if (rate == publicPresaleRate) {
```

Comment:
we can still make the fallback function call one or another, to make it easier for people who want to use that way